src/nnpack/nnlib.jl

```diff
@@ -0,0 +1,119 @@
+#NOTE: Commenting out the activation functions until sure what to do
+
+# relu(x::AA1) = nnp_relu_output(x, inplace ? x : similar(x), threadpool = shared_threadpool)
+
+# leakyrelu(x::AA1, a = oftype(x/1, 0.01)) =
+#     nnp_relu_output(x, inplace ? x : similar(x), negative_slope = a, threadpool = shared_threadpool)
+
+softmax!(x::A) where A<:AbstractVecOrMat{Float64} = softmax!(Float32.(x))
+
+softmax!(x::A) where A<:AbstractVecOrMat{Float32} =
+    nnp_softmax_output(x, x, threadpool = shared_threadpool)
+
+softmax!(y::A, x::A) where A<:AbstractVecOrMat{Float64} = softmax!(Float32.(y), Float32.(x))
+
+softmax!(y::A, x::A) where A<:AbstractVecOrMat{Float32} =
+    nnp_softmax_output(x, y, threadpool = shared_threadpool)
+
+softmax(x::A) where A<:AbstractVecOrMat{Float64} = softmax(Float32.(x))
+
+softmax(x::A) where A<:AbstractVecOrMat{Float32} =
```

Comment:
Doesn't NNlib effectively already have this definition?

Reply:
Yes it kind of does. But we do call into NNPACK softmax for all cases.